### ISSUE TITLE
What Does "this.db.open" Do?

### ISSUE THREAD

**Issue by JimTheMan** [ngrx/db#14]
So even though there is absolutely no documentation for this library, people are still using it. For example, https://github.com/ngrx/example-app/blob/master/src/app/effects/collection.ts

In that file there is this line:
```
 @Effect({ dispatch: false })
  openDB$: Observable<any> = defer(() => {
    return this.db.open('books_app');
  });
```

What the heck, man. This is unlike any @Effect I have ever seen. It doesn't check for an action, it uses some weird "defer" method, and it seems that into this.db.open you pass some string that it never used anywhere else in the app.

When is t he code actually executed? Is this run every time an action is dispatched? Thanks. : |

**Comment by mxchange** [comment]
I agree it's weird, that the example app uses this db thing that is not finished, and feels to me goes against using the store, or reducers to do things like `this.db.insert` in the effect. While as I understand it this is for offline storage, this should rather be a composable plugin similar to: `ngrx-store-localstorage`

**Comment by snapnurse-joshuamatthews** [comment]
Also this line is the only one giving me issue after porting this to nativescript. It doesn't throw any errors, but it hangs up the entire redux pipeline and no further actions are handled.

**Comment by michaeljota** [comment]
@snapnurse-joshuamatthews Maybe NativeScript is missing the required API for this to work. IDK. You shall try  [rxdb](https://github.com/pubkey/rxdb). I do not endorse or develop that tool, just fell that is what this could be and is not.

**Comment by Sparker73** [comment]
Guys, a course on ngrx explained to me a lot of undocumented things and I now understand a bit more on ngrx, it is really cheap and very instructive. I don't like the voice of the teacher but he teaches well I guess, lets hope you don't get asleep listening to the course. Here it is: [NgRx In Depth (Angular 6 and NgRx 6, with FREE E-Book)](https://www.udemy.com/ngrx-course/)
